### PR TITLE
Cody: ✨ Suggest follow-up topics

### PR DESCRIPTION
**Pull request by philipp-spiess**
This adds a new recipe that is used to suggest up to three follow-up topics for Cody. The recipe is executed with every user chat message but will not wait for the answer so that the experience is better (this is in line with comparable chat bots).

## ToDo

- [x] Add behind a feature flag for now 

## Test plan

https://user-images.githubusercontent.com/458591/234884949-cd71893a-ee12-408f-8d7f-b6ca76497b66.mov

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (+0.08 kb) | 0.00% (+0.08 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/ps/cody-next-question/compare-report.html;tab=live_object) for a full comparison between the commits 556084a and faf80a2 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/cody-shared/BUILD.bazel [hunk 1]
```diff
@@ -38,6 +38,8 @@ ts_project(
         "src/chat/recipes/helpers.ts",
         "src/chat/recipes/improve-variable-names.ts",
         "src/chat/recipes/langs.ts",
+        "src/chat/recipes/next-questions.ts",
```
Comment (abeatrix) [inline client/cody-shared/BUILD.bazel:41]: noob question: what does this change do? 😅

**Review by abeatrix — APPROVED** [review]
It looks so nice, love the button styles! 

Not sure if it's related to the changes made in this PR but the edit button now has a background color (still looks nice tho XD)

image


nit: Also tested on a popular light theme, maybe we can add a darker shade to the buttons on light theme?

image

**Comment by abeatrix** [comment]
Add to Changelog? 😏

File: client/cody-shared/src/chat/recipes/next-questions.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import { Interaction } from '../transcript/interaction'
+
+import { Recipe, RecipeContext } from './recipe'
+
+export class NextQuestions implements Recipe {
+    public id = 'next-questions'
+
+    public async getInteraction(humanChatInput: string, context: RecipeContext): Promise<Interaction | null> {
+        const selection = context.editor.getActiveTextEditorSelectionOrEntireFile()
+        if (!selection) {
```
[the remark this answers is not in the document]
Comment (philipp-spiess) [reply client/cody-shared/src/chat/recipes/next-questions.ts:17]: Oh yeah totes.

File: client/cody-shared/BUILD.bazel (hunk 1 above)
Comment (abeatrix) [reply client/cody-shared/BUILD.bazel:41]: I wasn't aware of that but this is good to know, tysm :D

**Comment by philipp-spiess** [comment]
@abeatrix this is the default vscode theme for secondary buttons, i'd rather not change anything there? 😅  The same styling that is used for e.g. the context button

**Comment by philipp-spiess** [comment]
@abeatrix @eseliger Do you have ideas for the feature flag name? Unfortunately we already use `suggestions` for the inline completions feature so I would pick something that's unrelated to this 😅 `keepConversationsGoing`? Lol I need your help

**Comment by philipp-spiess** [comment]
Added a feature flag and changelog entry. Any objections to shipping this today? Would be cool to get this out before my long weekend 🙂

**Review by eseliger — APPROVED** [review]

**Comment by eseliger** [comment]
> Added a feature flag and changelog entry. Any objections to shipping this today? Would be cool to get this out before my long weekend 🙂

Let's go!

**Pull request merged**